src/mame/video/k057714.h

```diff
@@ -9,9 +9,20 @@
 class k057714_device : public device_t
 {
 public:
+	// construction/destruction
+	template <typename T>
+	k057714_device(const machine_config& mconfig, const char* tag, device_t* owner, u32 clock, T&& screen_tag)
+		: k057714_device(mconfig, tag, owner, clock)
+	{
+		m_screen.set_tag(std::forward<T>(screen_tag));
+	}
+
 	k057714_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock);
+
 	auto irq_callback() { return m_irq.bind(); }
 
+	void set_pixclock(const XTAL &xtal);
```

Comment:
Does this chip have a separate bus clock and pixel clock?  The device clock doesn’t seem to be used for anything right now.  Is there a reason not to use the device clock for the pixel clock and implement `device_clock_changed` to catch on-the-fly changes?

Reply:
Asynchronous host I/O isn’t difficult to implement.  There are plenty of chips that don’t have a bus clock input.  You can just latch on the edge of the R/W signals.  Not saying that‘s the case here, but it’s not at all unusual.  Graphics chips driving internal operations off the pixel clock isn’t unusual either.

Reply:
I personally can't answer that question (no hardware). DragonMinded does have the hardware though so if she wants to check that out then I can change the implementation as required.

I did consider using the bus clock instead of setting the clock like this but flip flopped since I have no idea what is most correct to the hardware.